File: common/src/main/java/com/ssblur/alchimiae/block/AlchimiaeBlocks.java
package com.ssblur.alchimiae.block;

import com.ssblur.alchimiae.AlchimiaeMod;
import dev.architectury.registry.registries.DeferredRegister;
import dev.architectury.registry.registries.RegistrySupplier;
import net.minecraft.core.registries.Registries;
import net.minecraft.world.level.block.Block;

public class AlchimiaeBlocks {
  public static final DeferredRegister<Block> BLOCKS = DeferredRegister.create(AlchimiaeMod.MOD_ID, Registries.BLOCK);

  public static final RegistrySupplier<Block> BOILER = BLOCKS.register("boiler", BoilerBlock::new);

  public static void register() {
    BLOCKS.register();
  }
}

File: common/src/main/kotlin/com/ssblur/alchimiae/block/AlchimiaeBlocks.kt
package com.ssblur.alchimiae.block

import com.ssblur.alchimiae.AlchimiaeMod
import com.ssblur.alchimiae.item.AlchimiaeItems
import com.ssblur.unfocused.tab.CreativeTabs.tab

object AlchimiaeBlocks {
  val BOILER = AlchimiaeMod.registerBlockWithItem("boiler") { BoilerBlock() }
  val ALEMBIC = AlchimiaeMod.registerBlockWithItem("alembic") { AlembicBlock() }
  val FILTER = AlchimiaeMod.registerBlockWithItem("filter") { FilterBlock() }

  fun register() {
    BOILER.second.tab(AlchimiaeItems.TAB)
    ALEMBIC.second.tab(AlchimiaeItems.TAB)
    FILTER.second.tab(AlchimiaeItems.TAB)
  }
}
